org.adempiere.base/src/org/compiere/model/MSequence.java

```diff
@@ -564,6 +589,48 @@ public static String getDocumentNoFromSeq(MSequence seq, String trxName, PO po)
 				+ " - Sequence=" + AD_Sequence_ID + " [" + trx + "]");
 		return documentNo;
 	}
+	
+	private Boolean isSequenceNoLevel = null;
+	private Boolean isUsePrefixAsKey = null;
+	private Boolean isUseSuffixAsKey = null;
```

Comment:
this is not thread safe, use AtomicBoolean instead, i.e "private AtomicBoolean isSequenceNoLevel = new AtomicBoolean(false);"